src/hpa.c

```diff
@@ -378,6 +378,18 @@ static bool
 hpa_try_purge(tsdn_t *tsdn, hpa_shard_t *shard) {
 	malloc_mutex_assert_owner(tsdn, &shard->mtx);
 
+	/*
+	 * Make sure we respect purge interval setting and don't purge
```

Comment:
@ilvokhin to double check, it looks before this PR, the min interval is used to determine the next bg thd wakeup here:
https://github.com/jemalloc/jemalloc/blob/90c627edb70e081e1298b79010478d2f804467f1/src/hpa.c#L943

but we can still wake up early once we reach the early wakeup threshold (that is 4M)?

Reply:
Yes, this is matches my understanding as well. In addition, we might wake up early due to dirty_decay_ms even without early wakeup logic on the classic side.

In other words, we will still wake up background thread at whatever times current logic say us to wake up, but we will not purge anything, because of this condition.

Reply:
> we will still wake up background thread at whatever times current logic say us to wake up, but we will not purge anything

In that case, the bg thds waking up will only purge the PAC side of things, is that right? Because otherwise we might as well just enforce the next wake up, instead of still letting it wake up and then skip.

Reply:
> In that case, the bg thds waking up will only purge the PAC side of things, is that right?

Yes, exactly. We'll still purge PAC side.